probe_scraper/runner.py

```diff
@@ -438,30 +455,30 @@ def setup_output_and_cache_dirs(output_bucket, cache_bucket, out_dir, cache_dir)
 
 
 def sync_output_and_cache_dirs(
-    output_bucket, cache_bucket, out_dir, cache_dir, cache_path
+    output_bucket: str,
+    cache_bucket: str,
+    out_dir: Path,
+    cache_dir: Path,
+    cache_path: str,
 ):
     # Check output dir and then sync with cloudfront
     if not os.listdir(out_dir):
         print("{} is empty".format(out_dir))
         sys.exit(1)
     else:
-        print("Syncing output dir {}/ with s3://{}/".format(out_dir, output_bucket))
+        print(f"Syncing output dir {out_dir}/ with s3://{output_bucket}/")
 
         # cloudfront is supposed to automatically gzip objects, but it won't do that
         # if the object size is > 10 megabytes (https://webmasters.stackexchange.com/a/111734)
         # which our files sometimes are. to work around this, we'll regzip the contents into a
         # temporary directory, and upload that with a special content encoding
         with tempfile.TemporaryDirectory() as tmpdirname:
-            for root, dirnames, filenames in os.walk(out_dir):
-                rel_root = os.path.relpath(root, start=out_dir)
-                for dirname in dirnames:
-                    os.mkdir(os.path.join(tmpdirname, rel_root, dirname))
-                for filename in filenames:
-                    in_filename = os.path.join(root, filename)
-                    out_filename = os.path.join(tmpdirname, rel_root, filename)
-                    with open(in_filename, "rb") as f1:
-                        with gzip.open(out_filename, "wb") as f2:
-                            f2.write(f1.read())
+            tmp = Path(tmpdirname)
+            for in_filename in out_dir.rglob("*"):
+                if not in_filename.is_dir():
+                    out_filename = tmp / in_filename.relative_to(out_dir)
+                    out_filename.parent.mkdir(parents=True, exist_ok=True)
+                    out_filename.write_bytes(gzip.compress(in_filename.read_bytes()))
```

Comment:
I can't speak to this reconstruction. It appears as though it'll do the same thing (recursively globbing out_dir, and for each file ensuring its parent dirs are mirrored appropriately, then outputting gzipped bytes), but I'm not sufficiently familiar with the quirks of os.path and pathlib to be certain that there aren't dragons